### ISSUE TITLE
Can you provide specific version information of the environment or is the configuration file in the Config file correct?

### ISSUE THREAD

**Issue by jianfengzhai** [ayanban011/SwinDocSegmenter#2]
Hi,
Can you provide specific version information of the environment or is the configuration file in the Config file correct?

Are the configurations of the datasets in the Model Zoo chapter the same？all get in https://github.com/ayanban011/SwinDocSegmenter/blob/main/configs/coco/instance-segmentation/maskdino_R50_bs16_50ep_4s_dowsample1_2048.yaml？

I encountered some issues in Step 3: For testing the model, I loaded the model and configuration files, but encountered warning messages. I think it may be due to version issues, or whether these warning messages are normal. It seems that the forward predict results I received are also abnormal.
warnings as follows:
Environment info:
-------------------------------  -----------------------------------------------------------------------------
sys.platform                     linux
Python                           3.8.16 (default, Jun 12 2023, 18:09:05) [GCC 11.2.0]
numpy                            1.24.3
detectron2                       0.6 
Compiler                         GCC 5.5
CUDA compiler                    CUDA 11.2
detectron2 arch flags            7.5
DETECTRON2_ENV_MODULE            <not set>
PyTorch                          1.10.2 
PyTorch debug build              False
torch._C._GLIBCXX_USE_CXX11_ABI  False
GPU available                    Yes
GPU 0                            GeForce RTX 2080 Ti (arch=7.5)
Driver version                   460.27.04
CUDA_HOME                        /usr/local/cuda-11.2
Pillow                           9.4.0
torchvision                      0.11.3 
torchvision arch flags           3.5, 5.0, 6.0, 7.0, 7.5, 8.0, 8.6
fvcore                           0.1.5.post20221221
iopath                           0.1.9
cv2                              4.7.0
-------------------------------  -----------------------------------------------------------------------------
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2023.1-Product Build 20230303 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.2.3 (Git Hash 7336ca9f055cf1bfa13efb658fe15dc9b41f0740)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX512
  - CUDA Runtime 11.3
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.2
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.3, CUDNN_VERSION=8.2.0, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.10.2, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

[07/17 17:27:45 detectron2]: Command line arguments: Namespace(EVAL_FLAG=1, config_file='configs/coco/instance-segmentation/maskdino_R50_bs16_50ep_4s_dowsample1_2048_doc_publaynet.yaml', dist_url='tcp://127.0.0.1:10791', eval_only=True, machine_rank=0, num_gpus=1, num_machines=1, opts=['MODEL.WEIGHTS', 'weights/model_final_publay_swindocseg.pth'], resume=False)
[07/17 17:27:45 detectron2]: Contents of args.config_file=configs/coco/instance-segmentation/maskdino_R50_bs16_50ep_4s_dowsample1_2048_doc_publaynet.yaml:
_BASE_: Base-COCO-InstanceSegmentation.yaml
MODEL:
  META_ARCHITECTURE: "MaskDINO"
  SEM_SEG_HEAD:
    NAME: "MaskDINOHead"
    IGNORE_VALUE: 255
    NUM_CLASSES: 5 # 11
    LOSS_WEIGHT: 1.0
    CONVS_DIM: 256
    MASK_DIM: 256
    NORM: "GN"
    # pixel decoder
    PIXEL_DECODER_NAME: "MaskDINOEncoder"
    DIM_FEEDFORWARD: 2048
    NUM_FEATURE_LEVELS: 3
    TOTAL_NUM_FEATURE_LEVELS: 4
    IN_FEATURES: ["res2", "res3", "res4", "res5"]
    DEFORMABLE_TRANSFORMER_ENCODER_IN_FEATURES: ["res2","res3", "res4", "res5"]
    COMMON_STRIDE: 4
    TRANSFORMER_ENC_LAYERS: 6
    FEATURE_ORDER: "low2high"
  MaskDINO:
    TRANSFORMER_DECODER_NAME: "MaskDINODecoder"
    DEEP_SUPERVISION: True
    NO_OBJECT_WEIGHT: 0.1
    CLASS_WEIGHT: 4.0
    MASK_WEIGHT: 5.0
    DICE_WEIGHT: 5.0
    BOX_WEIGHT: 5.0
    GIOU_WEIGHT: 2.0
    HIDDEN_DIM: 256
    NUM_OBJECT_QUERIES: 300
    NHEADS: 8
    DROPOUT: 0.0
    DIM_FEEDFORWARD: 1024
    ENC_LAYERS: 0
    PRE_NORM: False
    ENFORCE_INPUT_PROJ: False
    SIZE_DIVISIBILITY: 32
    DEC_LAYERS: 9  # 9+1, 9 decoder layers, add one for the loss on learnable query
    TRAIN_NUM_POINTS: 12544
    OVERSAMPLE_RATIO: 3.0
    IMPORTANCE_SAMPLE_RATIO: 0.75
    EVAL_FLAG: 1
    INITIAL_PRED: True
    TWO_STAGE: True
    DN: "seg"
    DN_NUM: 100
    INITIALIZE_BOX_TYPE: 'no'
    TEST:
      SEMANTIC_ON: False
      INSTANCE_ON: True
      PANOPTIC_ON: False
      OVERLAP_THRESHOLD: 0.8
      OBJECT_MASK_THRESHOLD: 0.25
DATASETS:
#  TRAIN: ("doc_train",)
#  TEST: ("doc_val",)
  TEST: ("publaynet_minival",)
SOLVER:
  AMP:
    ENABLED: True
INPUT:
  IMAGE_SIZE: 512
TEST:
  EVAL_PERIOD: 5000
#  EVAL_FLAG: 1
OUTPUT_DIR: output/doc_publaynet 

**[07/17 17:27:58 d2.checkpoint.detection_checkpoint]: [DetectionCheckpointer] Loading from weights/model_final_publay_swindocseg.pth ...
[07/17 17:27:58 fvcore.common.checkpoint]: [Checkpointer] Loading from weights/model_final_publay_swindocseg.pth ...
WARNING [07/17 17:28:00 fvcore.common.checkpoint]: Skip loading parameter 'sem_seg_head.pixel_decoder.input_proj.0.0.weight' to the model due to **_incompatible shapes: (256, 192, 1, 1) in the checkpoint but (256, 256, 1, 1) in the model! You might want to double check if this is expected.
WARNING [07/17 17:28:00 fvcore.common.checkpoint]: Skip loading parameter 'sem_seg_head.pixel_decoder.input_proj.1.0.weight' to the model due to incompatible shapes: (256, 384, 1, 1) in the checkpoint but (256, 512, 1, 1) in the model! You might want to double check if this is expected.
WARNING [07/17 17:28:00 fvcore.common.checkpoint]: Skip loading parameter 'sem_seg_head.pixel_decoder.input_proj.2.0.weight' to the model due to incompatible shapes: (256, 768, 1, 1) in the checkpoint but (256, 1024, 1, 1) in the model! You might want to double check if this is expected.
WARNING [07/17 17:28:00 fvcore.common.checkpoint]: Skip loading parameter 'sem_seg_head.pixel_decoder.input_proj.3.0.weight' to the model due to incompatible shapes: (256, 1536, 1, 1) in the checkpoint but (256, 2048, 1, 1) in the model! You might want to double check if this is expected._**
WARNING [07/17 17:28:00 fvcore.common.checkpoint]: Skip loading parameter 'sem_seg_head.pixel_decoder.transformer.level_embed' to the model due to incompatible shapes: (5, 256) in the checkpoint but (4, 256) in the model! You might want to double check if this is expected.
WARNING [07/17 17:28:00 fvcore.common.checkpoint]: Skip loading parameter 'sem_seg_head.pixel_decoder.transformer.encoder.layers.0.self_attn.sampling_offsets.weight' to the model due to incompatible shapes: (320, 256) in the checkpoint but (256, 256) in the model! You might want to double check if this is expected.
WARNING [07/17 17:28:00 fvcore.common.checkpoint]: Skip loading parameter 'sem_seg_head.pixel_decoder.transformer.encoder.layers.0.self_attn.sampling_offsets.bias' to the model due to incompatible shapes: (320,) in the checkpoint but (256,) in the model! You might want to double check if this is expected.
WARNING [07/17 17:28:00 fvcore.common.checkpoint]: Skip loading parameter 'sem_seg_head.pixel_decoder.transformer.encoder.layers.0.self_attn.attention_weights.weight' to the model due to incompatible shapes: (160, 256) in the checkpoint but (128, 256) in the model! You might want to double check if this is expected.
WARNING [07/17 17:28:00 fvcore.common.checkpoint]: Skip loading parameter 'sem_seg_head.pixel_decoder.transformer.encoder.layers.0.self_attn.attention_weights.bias' to the model due to incompatible shapes: (160,) in the checkpoint but (128,) in the model! You might want to double check if this is expected.
WARNING [07/17 17:28:00 fvcore.common.checkpoint]: Skip loading parameter 'sem_seg_head.pixel_decoder.transformer.encoder.layers.0.linear1.weight' to the model due to incompatible shapes: (512, 256) in the checkpoint but (2048, 256) in the model! You might want to double check if this is expected.
WARNING [07/17 17:28:00 fvcore.common.checkpoint]: Skip loading parameter 'sem_seg_head.pixel_decoder.transformer.encoder.layers.0.linear1.bias' to the model due to incompatible shapes: (512,) in the checkpoint but (2048,) in the model! You might want to double check if this is expected.
WARNING [07/17 17:28:00 fvcore.common.checkpoint]: Skip loading parameter 'sem_seg_head.pixel_decoder.transformer.encoder.layers.0.linear2.weight' to the model due to incompatible shapes: (256, 512) in the checkpoint but (256, 2048) in the model! You might want to double check if this is expected.
WARNING [07/17 17:28:00 fvcore.common.checkpoint]: Skip loading parameter 'sem_seg_head.pixel_decoder.transformer.encoder.layers.1.self_attn.sampling_offsets.weight' to the model due to incompatible shapes: (320, 256) in the checkpoint but (256, 256) in the model! You might want to double check if this is expected.
WARNING [07/17 17:28:00 fvcore.common.checkpoint]: Skip loading parameter 'sem_seg_head.pixel_decoder.transformer.encoder.layers.1.self_attn.sampling_offsets.bias' to the model due to incompatible shapes: (320,) in the checkpoint but (256,) in the model! You might want to double check if this is expected.
WARNING [07/17 17:28:00 fvcore.common.checkpoint]: Skip loading parameter 'sem_seg_head.pixel_decoder.transformer.encoder.layers.1.self_attn.attention_weights.weight' to the model due to incompatible shapes: (160, 256) in the checkpoint but (128, 256) in the model! You might want to double check if this is expected.
WARNING [07/17 17:28:00 fvcore.common.checkpoint]: Skip loading parameter 'sem_seg_head.pixel_decoder.transformer.encoder.layers.1.self_attn.attention_weights.bias' to the model due to incompatible shapes: (160,) in the checkpoint but (128,) in the model! You might want to double check if this is expected.
WARNING [07/17 17:28:00 fvcore.common.checkpoint]: Skip loading parameter 'sem_seg_head.pixel_decoder.transformer.encoder.layers.1.linear1.weight' to the model due to incompatible shapes: (512, 256) in the checkpoint but (2048, 256) in the model! You might want to double check if this is expected.
WARNING [07/17 17:28:00 fvcore.common.checkpoint]: Skip loading parameter 'sem_seg_head.pixel_decoder.transformer.encoder.layers.1.linear1.bias' to the model due to incompatible shapes: (512,) in the checkpoint but (2048,) in the model! You might want to double check if this is expected.
WARNING [07/17 17:28:00 fvcore.common.checkpoint]: Skip loading parameter 'sem_seg_head.pixel_decoder.transformer.encoder.layers.1.linear2.weight' to the model due to incompatible shapes: (256, 512) in the checkpoint but (256, 2048) in the model! You might want to double check if this is expected.
WARNING [07/17 17:28:00 fvcore.common.checkpoint]:**

Could you give me some advice or helpful information about the problem ? thank you!

**Comment by ayanban011** [comment]
Updated Config files. The warnings are due to the image size mismatch.

**Comment by TimHunt12** [comment]
@jianfengzhai jianfengzhai I am running into the same issue and I see that @ayanban011 advised you to modify the config files to fix the image size mismatch.  I have been looking through the .yaml file and I am not sure what values I need to change.  Any suggestions?

```
_BASE_: Base-COCO-InstanceSegmentation.yaml
MODEL:
  META_ARCHITECTURE: "MaskDINO"
  SEM_SEG_HEAD:
    NAME: "MaskDINOHead"
    IGNORE_VALUE: 255
    NUM_CLASSES: 11
    LOSS_WEIGHT: 1.0
    CONVS_DIM: 256
    MASK_DIM: 256
    NORM: "GN"
    # pixel decoder
    PIXEL_DECODER_NAME: "MaskDINOEncoder"
    DIM_FEEDFORWARD: 2048
    NUM_FEATURE_LEVELS: 3
    TOTAL_NUM_FEATURE_LEVELS: 4
    IN_FEATURES: ["res2", "res3", "res4", "res5"]
    DEFORMABLE_TRANSFORMER_ENCODER_IN_FEATURES: ["res2","res3", "res4", "res5"]
    COMMON_STRIDE: 4
    TRANSFORMER_ENC_LAYERS: 6
    FEATURE_ORDER: "low2high"
  MaskDINO:
    TRANSFORMER_DECODER_NAME: "MaskDINODecoder"
    DEEP_SUPERVISION: True
    NO_OBJECT_WEIGHT: 0.1
    CLASS_WEIGHT: 4.0
    MASK_WEIGHT: 5.0
    DICE_WEIGHT: 5.0
    BOX_WEIGHT: 5.0
    GIOU_WEIGHT: 2.0
    HIDDEN_DIM: 256
    NUM_OBJECT_QUERIES: 300
    NHEADS: 8
    DROPOUT: 0.0
    DIM_FEEDFORWARD: 1024
    ENC_LAYERS: 0
    PRE_NORM: False
    ENFORCE_INPUT_PROJ: False
    SIZE_DIVISIBILITY: 32
    DEC_LAYERS: 9  # 9+1, 9 decoder layers, add one for the loss on learnable query
    TRAIN_NUM_POINTS: 12544
    OVERSAMPLE_RATIO: 3.0
    IMPORTANCE_SAMPLE_RATIO: 0.75
    EVAL_FLAG: 1
    INITIAL_PRED: True
    TWO_STAGE: True
    DN: "seg"
    DN_NUM: 100
    INITIALIZE_BOX_TYPE: 'no'
    TEST:
      SEMANTIC_ON: False
      INSTANCE_ON: True
      PANOPTIC_ON: False
      OVERLAP_THRESHOLD: 0.8
      OBJECT_MASK_THRESHOLD: 0.25
DATASETS:
  TRAIN: ("doc_train",)
  TEST: ("doc_val",)
SOLVER:
  AMP:
    ENABLED: True
TEST:
  EVAL_PERIOD: 5000
#  EVAL_FLAG: 1
OUTPUT_DIR: "/home/ubuntu/git/SwinDocSegmenter/data2/output_doclay_comp_Res"
```

**Comment by ayanban011** [comment]
Please change the DIM_FEEDFORWARD value of the SEM_SEG_HEAD as well as MasKDINO  based on your current input image size

**Comment by TimHunt12** [comment]
Well I answered my own question above.  It turns out that with all the .YAML files if I switch to the .YAML file that matches the model weights then the image sizes are correct and it works with all warning messages gone.
